packages/lib/src/components/ApplePay/ApplePayService.ts

```diff
@@ -66,8 +66,8 @@ class ApplePayService {
      */
     onpaymentauthorized(event: ApplePayJS.ApplePayPaymentAuthorizedEvent, onPaymentAuthorized) {
         return new Promise((resolve, reject) => onPaymentAuthorized(resolve, reject, event))
-            .then(() => {
-                this.session.completePayment(ApplePaySession.STATUS_SUCCESS);
+            .then(response => {
+                this.session.completePayment(response);
             })
             .catch(() => {
                 this.session.completePayment(ApplePaySession.STATUS_FAILURE);
```

Comment:
If we are talking about passing the customized errors property, wouldn't make more sense for them to be passed in the catch block?

Reply:
Using the catch block can also work. As long as we are giving the possibility - in any way- to send a customized result. 

But I am not sure, would it not be also a breaking chang if we are using the catch block instead ?

Reply:
I just asked to brainstorm a bit where to handle the errors - but yea, it will be a breaking change if implemented as you proposed